### PR TITLE
[ISSUE#8653]Fix health check bug

### PR DESCRIPTION
**Pull request by hujun-w-2**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

for #8653

## Brief changelog

1.持久化实例新增change操作
2.在jraft apply操作中校验client是否还存在，如果存在才更新实例

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 1]
```diff
@@ -180,6 +208,12 @@ public Response onApply(WriteRequest request) {
         }
     }
     
+    private void onInstanceChange(Service service, Instance instance, String clientId) {
+        if (clientManager.contains(clientId)) {
```
Comment (KomachiSion) [inline naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:212]: Whether this judgment is accurate？

Comment (hujun-w-2) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:212]: The solution here is that if the instance is logged out, the updated instance will not re register the instance. The logged out "apply" will remove the client, and raft can ensure the order of the logged out "apply" and the updated "apply"

> Whether this judgment is accurate？

Comment (KomachiSion) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:212]: How about one client register two persistent instance, but only remove one instance?

Comment (hujun-w-2) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java]: i will fix it

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 2]
```diff
@@ -170,6 +195,12 @@ public Response onApply(WriteRequest request) {
                 case DELETE:
                     onInstanceDeregister(instanceRequest.service, instanceRequest.getClientId());
                     break;
+                case CHANGE:
+                    if (instanceAndServiceExist(instanceRequest)) {
```
Comment (pixystone) [inline naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:199]: Discussion: would be better an atomic operation?

**Review by pixystone — CHANGES_REQUESTED** [review]

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java [hunk 3]
```diff
@@ -118,6 +118,31 @@ public void registerInstance(Service service, Instance instance, String clientId
...
+            throw new NacosRuntimeException(NacosException.INVALID_PARAM,
+                    String.format("Current service %s is ephemeral service, can't update persistent instance.",
+                            singleton.getGroupedServiceName()));
+        }
+        final PersistentClientOperationServiceImpl.InstanceStoreRequest request = new PersistentClientOperationServiceImpl.InstanceStoreRequest();
+        request.setService(service);
+        request.setInstance(instance);
+        request.setClientId(clientId);
+        final WriteRequest writeRequest = WriteRequest.newBuilder()
+                .setGroup(Constants.NAMING_PERSISTENT_SERVICE_GROUP_V2)
```
[the remark this answers is not in the document]
Comment (hujun-w-2) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:136]: ok

File: naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java (hunk 2 above)
Comment (hujun-w-2) [reply naming/src/main/java/com/alibaba/nacos/naming/core/v2/service/impl/PersistentClientOperationServiceImpl.java:199]: > Discussion: would be better an atomic operation?

The "log" of "jraft" is executed in sequence, and the next log will not be executed before it is executed, so I don't think it's a big problem here.

**Comment by KomachiSion** [comment]
coverage of `[PersistentClientOperationServiceImpl.java](https://codecov.io/gh/alibaba/nacos/pull/8639/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=alibaba#diff-bmFtaW5nL3NyYy9tYWluL2phdmEvY29tL2FsaWJhYmEvbmFjb3MvbmFtaW5nL2NvcmUvdjIvc2VydmljZS9pbXBsL1BlcnNpc3RlbnRDbGllbnRPcGVyYXRpb25TZXJ2aWNlSW1wbC5qYXZh)` reduced

**Comment by KomachiSion** [comment]
> coverage of [PersistentClientOperationServiceImpl.java](https://codecov.io/gh/alibaba/nacos/pull/8639/diff?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=alibaba#diff-bmFtaW5nL3NyYy9tYWluL2phdmEvY29tL2FsaWJhYmEvbmFjb3MvbmFtaW5nL2NvcmUvdjIvc2VydmljZS9pbXBsL1BlcnNpc3RlbnRDbGllbnRPcGVyYXRpb25TZXJ2aWNlSW1wbC5qYXZh) reduced

**Review by KomachiSion — APPROVED** [review]

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**